### PR TITLE
Migrate `@woocmmerce/onboarding` to TS

### LINKED ISSUE #33132: Migrate `@woocommerce/onboarding` to TS
### Describe the solution you'd like

convert @woocommerce/customer-effort-score to TypeScript and output type definitions

### Describe alternatives you've considered

_No response_

### Additional context

Part of https://github.com/woocommerce/woocommerce/issues/32179

### PR DESCRIPTION
**Pull request by chihsuan**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #33132

This PR migrates @woocmmerce/onboarding to TS and fixes the related type errors in ./client.

### How to test the changes in this Pull Request:

1. Run `pnpm nx build @woocmmerce/onboarding `
2. Observe that `packages/js/onboarding/build-types` is generated

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33153/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Add changelog` 637ea244eac80754dab8109679f804fb3d6504c0
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: packages/js/onboarding/src/components/WooOnboardingTask/WooOnboardingTask.tsx [hunk 1]
```diff
@@ -35,18 +35,29 @@ export const trackView = async ( taskId, variant ) => {
...
  * A Fill for adding Onboarding tasks.
  *
  * @slotFill WooOnboardingTask
  * @scope woocommerce-tasks
- * @param {Object} props          React props.
- * @param {string} props.variant  The variant of the task.
- * @param {Object} props.children React component children
- * @param {string} props.id       Task id.
+ * @param {Object} props           React props.
+ * @param {string} [props.variant] The variant of the task.
```
Comment (ilyasfoo) [inline packages/js/onboarding/src/components/WooOnboardingTask/WooOnboardingTask.tsx:54]: I don't understand the `[]` add here, @chihsuan. Can you shed a light? Thanks!

**Review by ilyasfoo — APPROVED** [review]
Nice work, LGTM! I have a question but I think this is good to go 👍

File: packages/js/onboarding/src/components/WooOnboardingTask/WooOnboardingTask.tsx (hunk 1 above)
Comment (chihsuan) [reply packages/js/onboarding/src/components/WooOnboardingTask/WooOnboardingTask.tsx:54]: It indicates that `variant` is optional.

https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

Comment (ilyasfoo) [reply packages/js/onboarding/src/components/WooOnboardingTask/WooOnboardingTask.tsx:54]: TIL. Thanks!

**Pull request merged**